peekingduck/pipeline/nodes/input/visual.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2022 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Reads inputs from multiple visual sources
+- image or video file on local storage
+- folder of images or videos
+- online cloud source
+- CCTV or webcam live feed
+"""
+
+from pathlib import Path
+from typing import Any, Dict, Optional, Union
+
+from peekingduck.pipeline.nodes.input.utils.preprocess import resize_image
+from peekingduck.pipeline.nodes.input.utils.read import VideoNoThread, VideoThread
+from peekingduck.pipeline.nodes.node import AbstractNode
+
+
+class Node(AbstractNode):  # pylint: disable=too-many-instance-attributes
+    """Receives visual sources as inputs.
+
+    Inputs:
+        |none_input_data|
+
+    Outputs:
+        |img_data|
+
+        |filename_data|
+
+        |pipeline_end_data|
+
+        |saved_video_fps_data|
+
+    Configs:
+        filename (:obj:`str`): **default = "video.mp4"**. |br|
+            Filename of the MP4 file if media is exported. |br|
+            If source is a directory of files, then filename is the current file being
+            processed.
```

Comment:
1. Suggest making it clear that this is only valid if the input is a live stream, and for recorded videos it doesn't change the name.
2. Also suggest in `output.media_writer` docstring to add that a timestamp will be added to the video name

Reply:
Done.
I changed `output.media_writer` timestamp to YYMMDD_HHMMSS (from DDMMYY_HH_MM_SS) so that it facilitates sorting and is shorter.